src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -473,6 +456,62 @@ public Long getAllowedTokenExpirationClockSkewSeconds() {
         return allowedTokenExpirationClockSkewSeconds;
     }
 
+    @PostConstruct
+    @Restricted(NoExternalUse.class)
+    public void createProxyAwareResourceRetriver() {
+        proxyAwareResourceRetriever =
+                ProxyAwareResourceRetriever.createProxyAwareResourceRetriver(isDisableSslVerification());
+    }
+
+    ProxyAwareResourceRetriever getResourceRetriever() {
+        return proxyAwareResourceRetriever;
+    }
+
+    private OidcConfiguration buildOidcConfiguration() {
+        // TODO cache this and use the well known if available.
```

Comment:
for a future PR.  We can cache the client and invalidate it every 1 hour if using the well-known (as the well known refresh is now hard coded to 1hr).